data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/S3ObjectWorker.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.source;
+
+import com.amazon.dataprepper.model.buffer.Buffer;
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.record.Record;
+import com.amazon.dataprepper.plugins.source.codec.Codec;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.ResponseInputStream;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+
+import java.io.IOException;
+
+/**
+ * Class responsible for taking an {@link S3ObjectReference} and creating all the necessary {@link Event}
+ * objects in the Data Prepper {@link Buffer}.
+ */
+class S3ObjectWorker {
+    private static final Logger LOG = LoggerFactory.getLogger(S3ObjectWorker.class);
+
+    private final S3Client s3Client;
+    private final Buffer<Record<Event>> buffer;
+    private final Codec codec;
+    private final int bufferTimeoutInMillis;
+    private final int recordsToAccumulate;
+
+    public S3ObjectWorker(final S3Client s3Client,
+                          final Buffer<Record<Event>> buffer,
+                          final Codec codec,
+                          final int bufferTimeoutInMillis,
+                          final int recordsToAccumulate) {
+        this.s3Client = s3Client;
+        this.buffer = buffer;
+        this.codec = codec;
+        this.bufferTimeoutInMillis = bufferTimeoutInMillis;
+        this.recordsToAccumulate = recordsToAccumulate;
+    }
+
+    void parseS3Object(final S3ObjectReference s3ObjectPointer) throws IOException {
+        final GetObjectRequest getObjectRequest = GetObjectRequest.builder()
+                .bucket(s3ObjectPointer.getBucketName())
+                .key(s3ObjectPointer.getKey())
+                .build();
+
+        final BufferAccumulator bufferAccumulator = BufferAccumulator.create(buffer, recordsToAccumulate, bufferTimeoutInMillis);
```

Comment:
Should this object be created in the constructor?

Reply:
This class is single-threaded, and we may eventually have multiple threads running. So I instantiate this here to prevent multiple threads conflicting.